Kin-Rewards-Engine/Reputation-Model-Proposal.md

```diff
@@ -0,0 +1,44 @@
+	* Whether the address is attached to the mobile app or not. In case it is, a lot of data is available - how long does the user have the app, how active is he, his age, occupation, etc. This can be calculated locally on the device and sent to a Kik server (to preserve privacy) or done on Kik's servers (if possible legally). E.g., the amount of addresses a certain device opened is super relevant to that user's reputation.
+ 	* Age of coins. The longer the coins sit in an address the more likely they are trustworthy.
+ 	* Age of address. E.g., a new address will have reputation 0 and will be completely ignored by the reward engine (unless it was opened in a wallet of a specific type of user). The age of the address will be calculated either using its creation time (when available; e.g., contracts) or when the first incoming transaction has happened.
+	* Interaction with many developers (an address interacting with plenty of developers is highly likely to be honest).
+	* And more...
+
+The final score of a developer will be calculated as a weighted sum of his reputation, his users' reputations, and a regularization parameter that makes sure too sudden changes aren't possible (a developer doesn't become super popular in one day).
+
+### Other Ideas to Consider
+
+1. Anti-fraud - Kik will analyze Kin's transactions' graph to find fraud, e.g. Kin tokens that are being transferred from a developer to himself or any other circular references. There are pretty effective clustering mechanisms which already exist on the Bitcoin blockchain, in addition to lots of fraud detection technique which will apply in our case as well. Not sure how effective they are on Ethereum/ERC20 tokens, but worth researching it.
+2. Frequency - We are not sure a daily distribution is the best way to go. The distribution should match the activity on the system in some sense. For example, we could say a distribution takes place once a specified amount of Kin was transferred (again, only by reputable users). The amount of the distribution can be determined according to the amount of days since the last one took place.
+3. Soft voting - in a soft voting mechanism a user can vote for many developers according to his interaction with each of them. In addition, the distribution of a user's vote among many developers should not be linear in order to minimize gaps and thrive to a more evenly structured vote. Finally, users normally have pretty repetitive patterns of interaction with such ecosystems and we can divide the daily sum a user has spent with his average spending on the platform to correct his weight in the reward formula (again, in order to minimize unwanted gaps between strong and early developers and weaker and newer ones).
```

Comment:
I would rename "developers" to "digital services". Doesn't necessarily have to be a developer.